### ISSUE TITLE
Unrecognized options "memory_cost, time_cost, threads" under "security.encoders.Symfony\Component\Security\Core\User\User"

### ISSUE THREAD

**Issue by etshy** [symfony/symfony#29351]
**Symfony version(s) affected**: during update to 4.1.8

**Description of the problem including expected versus actual behavior**:
tried to `composer update` my project and I had the message : `Unrecognized options "memory_cost, time_cost, threads" under "security.encoders.FOS\UserBundle\Model\UserInterface"` (my `security.encoders` config was on `FOS\UserBundle\Model\UserInterface` I then changed it to `Symfony\Component\Security\Core\User\User`)
Now I have an excpetion on my app with the message  : `Unrecognized options "memory_cost, time_cost, threads" under "security.encoders.Symfony\Component\Security\Core\User\User"`

**Steps to reproduce**:
 1. `composer update`
 2. "global" exception on my app

**Provide logs (if relevant)**:
During update, I got this error message
```
Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 1
!!  PHP Warning:  Module 'mongodb' already loaded in Unknown on line 0
!!
!!  In ArrayNode.php line 307:
!!
!!    Unrecognized options "memory_cost, time_cost, threads" under "security.enco
!!    ders.FOS\UserBundle\Model\UserInterface"
!!
!!
!!
Script @auto-scripts was called via post-update-cmd
```

I guess it's related to my password encoding config with argon2i (my config is the same as the docs)
```
security:
  encoders:
    Symfony\Component\Security\Core\User\User:
      algorithm: argon2i
      memory_cost:          16384 # Amount in KiB. 16 MiB
      time_cost:            2 # Number of iterations
      threads:              4 # Number of parallel threads
```

**Comment by ro0NL** [comment]
4.0

```
  Unrecognized options "memory_cost, time_cost, threads" under "security.encoders.boo"  
```

4.1

all good.

**Comment by etshy** [comment]
Here is my composer.json

```
{
    "type": "project",
    "license": "proprietary",
    "require": {
        "php": "^7.2",
        "ext-iconv": "*",
        "ext-json": "*",
        "alcaeus/mongo-php-adapter": "^1.1@dev",
        "claviska/simpleimage": "dev-master",
        "doctrine/doctrine-bundle": "^1.9",
        "doctrine/mongodb-odm-bundle": "^3.5@dev",
        "doctrine/orm": "^2.6",
        "friendsofsymfony/user-bundle": "^2.0",
        "knplabs/knp-time-bundle": "^1.1@dev",
        "sensio/framework-extra-bundle": "^5.2@dev",
        "sensiolabs/security-checker": "^4.1",
        "stof/doctrine-extensions-bundle": "^1.3@dev",
        "symfony/apache-pack": "dev-master",
        "symfony/asset": "^4.0",
        "symfony/console": "^4.0",
        "symfony/dotenv": "^4.0",
        "symfony/expression-language": "^4.0",
        "symfony/flex": "^1.0",
        "symfony/framework-bundle": "^4.0",
        "symfony/messenger": "^4.0",
        "symfony/psr-http-message-bridge": "^1.0@dev",
        "symfony/security-csrf": "^4.0",
        "symfony/swiftmailer-bundle": "^3.2@dev",
        "symfony/translation": "^4.0",
        "symfony/twig-bundle": "^4.0",
        "symfony/validator": "^4.0",
        "symfony/yaml": "^4.0",
        "white-october/pagerfanta-bundle": "^1.0@dev",
        "zendframework/zend-diactoros": "^1.8@dev"
    },
    "require-dev": {
        "symfony/profiler-pack": "dev-master",
        "symfony/web-server-bundle": "^4.0"
    },
    "minimum-stability": "stable",
    "config": {
        "preferred-install": {
            "*": "dist"
        },
        "sort-packages": true,
        "platform": {
            "ext-mongo": "1.6.16"
        }
    },
    "autoload": {
        "psr-4": {
            "App\\": "src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "App\\Tests\\": "tests/"
        }
    },
    "replace": {
        "symfony/polyfill-iconv": "*",
        "symfony/polyfill-php71": "*",
        "symfony/polyfill-php70": "*",
        "symfony/polyfill-php56": "*"
    },
    "scripts": {
        "auto-scripts": {
            "cache:clear": "symfony-cmd",
            "assets:install --symlink --relative %PUBLIC_DIR%": "symfony-cmd",
            "security-checker security:check": "script"
        },
        "post-install-cmd": [
            "@auto-scripts"
        ],
        "post-update-cmd": [
            "@auto-scripts"
        ]
    },
    "conflict": {
        "symfony/symfony": "*"
    },
    "extra": {
        "symfony": {
            "id": "01C3BFXW85JTBRN0EH626VXM88",
            "allow-contrib": false
        }
    }
}

```
Maybe some of the deps are still on 4.0, I'll check that later.

I'll check the version I have for all the symfony packages and post it as soon as I can.

**Comment by etshy** [comment]
` composer show symfony/security`
returns this : `versions : * v3.4.19`

I don't knwo why though,; I started the project with symfony 4.
How can I update security to 4.1 ?
I tried `composer update symfony/security` but got:
```
Loading composer repositories with package information
Updating dependencies (including require-dev)
Nothing to install or update
Generating autoload files
Executing script cache:clear [KO]
 [KO]
Script cache:clear returned with error code 1
!!  PHP Warning:  Module 'mongodb' already loaded in Unknown on line 0
!!
!!  In ArrayNode.php line 307:
!!
!!    Unrecognized options "memory_cost, time_cost, threads" under "security.enco
!!    ders.Symfony\Component\Security\Core\User\User"
!!
!!
!!
Script @auto-scripts was called via post-update-cmd
```

Nothing to update

**Comment by etshy** [comment]
Well, I don't know from which package I got the security-bundle in my app (maybe `symfony/security-csrf`? but this one was in v4.1.8...), as it's not in my composer.json, but I was stuck with v3.4.19

I tried to `composer require symfony/security-bundle` but as it was already installed it didn't worked.

I had to delete the composer.lock and delete the security's folder in my vendor/symfony directory.

And from there `composer require symfony/security-bundle` worked.
